src/Layers/xrRender/ParticleEffect.cpp

```diff
@@ -643,6 +643,16 @@ void CParticleEffect::Render(float)
     TAL_SCOPED_TASK_NAMED("CParticleEffect::Render()");
 #endif // _GPA_ENABLED
 
+#ifdef USE_OGL
+    // Due to the big impact on performance
+    Fvector dir;
+    dir.sub(RDEVICE.vCameraPosition, m_InitialPosition);
+    float dist = dir.magnitude();
+float distSQ = RDEVICE.vCameraPosition.distance_to_sqr(m_InitialPosition)+EPS;
+if (distSQ > _sqr(100.f*psVisDistance))
+        return;
+#endif
```

Comment:
The formatting is broken now and `dir` and `dist` variables are unused and should be removed.